### ISSUE TITLE
DietPi | Debian Buster

### ISSUE THREAD

**Issue by MichaIng** [MichaIng/DietPi#1286]
### VMware/VirtualBox Buster images: https://dietpi.com/downloads/images/
- _Approx release: Summer 2019_
- _Transition and soft freeze have been applied._
- _Full freeze planned on 13. March 2019: https://wiki.debian.org/DebianBuster_
___________________
### Buster VM (x64_86) rough install tests
🈯️ `dietpi-software install {1..23}`
🈯️ `dietpi-software install {24..40}` (-27, not supported on VM)
🈯️ `dietpi-software install {41..65}` (-47,51,59,60,61)
~🈴~ 🈯️ OpenBazaar (58)
```
# github.com/OpenBazaar/openbazaar-go
fatal error: runtime: out of memory
...
```
- Out of memory? Hmm after all those installs several services are running... Okay totally exceptional situation of course.
- Jep, memory usage raises up to 922M, this time reaches some next step where it then fails again:
```
# github.com/OpenBazaar/openbazaar-go
/usr/local/go/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
/usr/bin/ld: /tmp/go-link-270012792/go.o: error adding symbols: memory exhausted
collect2: error: ld returned 1 exit status
```
- Okay ~1.5G is required for OpenBazaar to compile. No Buster issue!

🈯️ `dietpi-software install {66..74} 81 86 90 91 92` (-66,69,70,71,72)
~🈴~ phpMyAdmin (90), APT package not available on Debian Buster. Raspbian Buster has it... **EDIT: Blocked on non-RPi Buster: https://github.com/MichaIng/DietPi/commit/ad38e8833dbc9e2aadc0f934e45a2755586fd713**
🈯️ `dietpi-software install {93..102} {106..120}` (minus non x86/VM supported)
🈯️ `dietpi-software install {121..150}` (minus non x86/VM supported)
~🈴~ NAA Daemon (124) wants to install `gcc-6-base` but on Buster `gcc-8-base` is available only.
~🈴~ Tomcat8 (125) not available on Buster (https://packages.debian.org/tomcat8), Tomcat9 instead: https://packages.debian.org/tomcat9 ... But again it IS available on Raspbian Buster (both, Tomcat8+9). **EDIT: Resolved with: https://github.com/MichaIng/DietPi/commit/ae701d89437128fff70f710c0cd03777dad0118c**
🈯️ `dietpi-software install {151..175}` (minus non x86/VM supported)
**Further (mostly Buster related) fixes:**
- https://github.com/MichaIng/DietPi/commit/672e53924494e5c2f70f8f4361a143165edc8512
- https://github.com/MichaIng/DietPi/commit/472c46af4de818f76a912ee946afb9e578cca240
- https://github.com/MichaIng/DietPi/commit/cb9206ecaaaed2eff2cc84bbcfc9c534f61d4a12
___________________
As I anyway created a third VirtualBox image as fresh Stretch, I just became interested in further upgrading to Debian Buster. It is never to early to watch changes there and how we can resolve them for DietPi:

**VirtualBox/General:**
1. 🈯️  **NTP** now is a systemd service, thus removing the init service file does not disable it anymore: https://github.com/Fourdee/DietPi/blob/master/PREP_SYSTEM_FOR_DIETPI.sh#L318-L320
`dietpi-services mask ntp` works: **Fixed with https://github.com/Fourdee/DietPi/pull/1422**
Other DietPi ntpd commands still work well, nothing to change here so far.
2. 🈯️  Distro index **'buster'** need to be recognized: https://github.com/Fourdee/DietPi/blob/master/dietpi/dietpi-obtain_hw_model#L618-L627 **=> fixed in between**
3. 🈯️ APT **`mysql-server`** does not even exist anymore on Buster at the moment. Good we get rid of it in Stretch: https://packages.debian.org/de/stretch/mysql-server **=> fixed with: https://github.com/Fourdee/DietPi/pull/1276**
4. 🈯️ Got another error on webserver stack installation: **`php-xsl`** does not exist, which we install for all non Apache stacks, but it does not even exist on Stretch: https://packages.debian.org/search?searchon=names&keywords=php-xsl
Wonder why we never watched APT error here, maybe new. `php5-xsl` on Jessie exists, `php7.0-xsl` on Stretch, but it is just a dummy, which is better covered by `php-xml`: https://packages.debian.org/stretch/php7.0-xsl
So we should just remove this already beginning with Stretch: https://github.com/Fourdee/DietPi/blob/testing/dietpi/dietpi-software#L3857-L3861 **=> Fixed with: https://github.com/Fourdee/DietPi/pull/1326**
5. 🈯️  Watch massive **systemd** error on reboot, but everything is functional and not a DietPi issue: https://github.com/systemd/systemd/issues/7054: **Fixed by systemd devs**
6. 🈯️  Issue with **ownCloud** installation: When AGI fails, due to non existing Buster ownCloud repo, then further installation stops. As repo does not provide anything but the archive anymore (compared to older situation where owncloud-deps forced apache2 mysql etc.), the best solution is to simply rely on direct archive download. Internal updater does a good job, thus apt repo is totally useless. **=> fixed with: https://github.com/Fourdee/DietPi/pull/1325**
7. 🈯️ Mosquitto: No Buster repo available! Official Debian repo could be used, but not current version: https://packages.debian.org/buster/mosquitto, also Stretch repo seams to work on Buster, at least apt installation and service startup worked fine during test: : **Fixed with https://github.com/Fourdee/DietPi/pull/1422**
8. 🈯️ <s>Deluge is not available in Debian Buster repo (but available on Raspbian Buster), thus fails to install</s> **€: Now availabe: https://packages.debian.org/buster/deluge**
9. 🈯️ timedatectl output changed, thus run_ntpd status fails: **=> fixed with: https://github.com/Fourdee/DietPi/pull/1661**

_______________

**RPi:**
1. 🈳 Current version 0.6 of `raspi-copies-and-fills` breaks Buster udev to start up and breaks core system/reboot by this. It is solvable by `apt purge raspi-copies-and-fills`, but then the system is missing the armhf optimized binaries, this package ships. **=> Clearly no advice to update to Buster for now!** Reference: https://raspberrypi.stackexchange.com/a/74406 **=> Workaround for now: https://github.com/Fourdee/DietPi/pull/1472**
**EDIT: Now `archive.raspberrypi.org` has a Buster branch, which does not contain the `raspi-copies-and-fills` package anymore.**
2. 🈯️ Buster now has PHP7.2 as default, which does not support php-mcrypt. **=> Fixed with: https://github.com/Fourdee/DietPi/pull/1478**
   - 🈯️ `libsodium23`, and PHP7.2 compatible `php-redis`, `php-igbinary` and `php-apcu` modules now available on Raspbian Buster repo as well.
   - 🈯️ <s>€: `php-redis`, `php-igbinary` and `php-apcu` PHP7.2 compatibility lost again... WTF???</s> **€: Again resolved!**
3. 🈯️ General info:

On Debian Buster repo there are many many package updates so far, and somehow the mirroring to Raspbian Buster repo somehow is not done well (see missing `libsodium23` above). At the moment there is some perl update, which would due to broken dependencies remove Apache and MariaDB (and for sure many other titels):
```
2018-02-20 00:26:39 root@micha:~# apt full-upgrade
Reading package lists... Done
Building dependency tree
Reading state information... Done
Calculating upgrade... Done
The following packages were automatically installed and are no longer required:
  apache2-data apache2-utils augeas-lenses certbot dh-python galera-3 libaio1 libapr1 libaprutil1 libaprutil1-dbd-sqlite3 libaprutil1-ldap libaugeas0 libbrotli1 libgdbm3 liblua5.2-0 libmpdec2 libperl5.26 libpython3-stdlib
  libpython3.6-minimal libpython3.6-stdlib libreadline5 libwrap0 lsof mariadb-client-core-10.1 mariadb-common mariadb-server-core-10.1 perl-modules-5.26 python3 python3-acme python3-asn1crypto python3-augeas python3-certbot
  python3-certifi python3-cffi-backend python3-chardet python3-configargparse python3-configobj python3-cryptography python3-distutils python3-future python3-idna python3-josepy python3-lib2to3 python3-minimal python3-mock
  python3-openssl python3-parsedatetime python3-pbr python3-pkg-resources python3-requests python3-rfc3339 python3-six python3-tz python3-urllib3 python3-zope.component python3-zope.event python3-zope.hookable python3-zope.interface
  python3.6 python3.6-minimal socat
Use 'apt autoremove' to remove them.
The following packages will be REMOVED:
  apache2 apache2-bin libapache2-mod-php libapache2-mod-php7.2 libconfig-inifiles-perl libdbi-perl mariadb-client-10.1 mariadb-server mariadb-server-10.1 mysqltuner perl python-certbot-apache python3-certbot-apache
The following packages will be upgraded:
  perl-base
1 upgraded, 0 newly installed, 13 to remove and 0 not upgraded.
Need to get 1,241 kB of archives.
After this operation, 123 MB disk space will be freed.
Do you want to continue? [Y/n] n
Abort.
2018-02-20 14:53:43 root@micha:~# apt list --upgradable
Listing... Done
perl/testing 5.26.1-4+b1 armhf [upgradable from: 5.26.1-4]
perl-base/testing 5.26.1-4+b1 armhf [upgradable from: 5.26.1-4]
```
As `dietpi-software` forces `apt-get dist-upgrade` on installation, it would just force this removal.

On Debian I didn't face such problems so far, it seems to be the transfer from Debian repo to Raspbian repo that is not done too carefully by times.</details> **Issue resolved on 2. Feb. 2018: https://github.com/Fourdee/DietPi/issues/1286#issuecomment-369408959**

4. 🈯️ Reset raspi.list "buster" to currently highest "stretch" in case: https://github.com/Fourdee/DietPi/commit/374c09dd5f2c8c2446c82411adc758708bdd9620

5. 🈯️ <s>Current repo inconsistency issue with Apache2 and PHP: https://github.com/Fourdee/DietPi/issues/1286#issuecomment-394041400</s> **Resolved!**
6. 🈯️ <s>Again udev restart (+journald and I guess other systemd services) failure after upgrade. Kernel update and systemd/udev update at the same time. Perhaps reboot solves it, but I am afraid to reboot, as server otherwise runs fine currently. Raspbian Buster systemd/udev is on v238 while Debian Buster is on v239. Will wait for next update, before attempting reboot.
  - Installing Debian armhf udev v239 allowed systemd-udevd to start up. However systemd-journald still fails, perhaps due to version incompatibility.</s> **€: Solved with Raspbian systemd+udev v239 APT upgrade**
7. 🈴 `i-sable-2km` module currently might fail on Buster due to `gcc-4.9` version mismatch (4.9.3 vs. 4.9.4): https://github.com/Fourdee/DietPi/issues/2547#issuecomment-463831296
Might still work, since e.g. `rpi-update` just checks `4.9` but not the sub sub version. Requires testing.

**Comment by Fourdee** [comment]
@MichaIng 

I've set low priority label for info (We officially support stretch), however, don't change what you like to work on, or by how much 👍

**Comment by MichaIng** [comment]
@Fourdee 
Jep, agree. Just fixed the issues I observed, as they were not too hard and I want to test/go with DietPi Buster on my RPi after v6.0 release, at least give it a try 😄.

**Comment by MichaIng** [comment]
🈴 General info: On Debian Buster repo there are many many package updates so far, and somehow the mirroring to Raspbian Buster repo is not done well (see missing `libsodium23` above). At the moment there is some perl update, which would remove Apache and MariaDB due to broken dependencies (and for sure many other titles):
```
2018-02-20 00:26:39 root@micha:~# apt full-upgrade
Reading package lists... Done
Building dependency tree
Reading state information... Done
Calculating upgrade... Done
The following packages were automatically installed and are no longer required:
  apache2-data apache2-utils augeas-lenses certbot dh-python galera-3 libaio1 libapr1 libaprutil1 libaprutil1-dbd-sqlite3 libaprutil1-ldap libaugeas0 libbrotli1 libgdbm3 liblua5.2-0 libmpdec2 libperl5.26 libpython3-stdlib
  libpython3.6-minimal libpython3.6-stdlib libreadline5 libwrap0 lsof mariadb-client-core-10.1 mariadb-common mariadb-server-core-10.1 perl-modules-5.26 python3 python3-acme python3-asn1crypto python3-augeas python3-certbot
  python3-certifi python3-cffi-backend python3-chardet python3-configargparse python3-configobj python3-cryptography python3-distutils python3-future python3-idna python3-josepy python3-lib2to3 python3-minimal python3-mock
  python3-openssl python3-parsedatetime python3-pbr python3-pkg-resources python3-requests python3-rfc3339 python3-six python3-tz python3-urllib3 python3-zope.component python3-zope.event python3-zope.hookable python3-zope.interface
  python3.6 python3.6-minimal socat
Use 'apt autoremove' to remove them.
The following packages will be REMOVED:
  apache2 apache2-bin libapache2-mod-php libapache2-mod-php7.2 libconfig-inifiles-perl libdbi-perl mariadb-client-10.1 mariadb-server mariadb-server-10.1 mysqltuner perl python-certbot-apache python3-certbot-apache
The following packages will be upgraded:
  perl-base
1 upgraded, 0 newly installed, 13 to remove and 0 not upgraded.
Need to get 1,241 kB of archives.
After this operation, 123 MB disk space will be freed.
Do you want to continue? [Y/n] n
Abort.
2018-02-20 14:53:43 root@micha:~# apt list --upgradable
Listing... Done
perl/testing 5.26.1-4+b1 armhf [upgradable from: 5.26.1-4]
perl-base/testing 5.26.1-4+b1 armhf [upgradable from: 5.26.1-4]
```
As `dietpi-software` forces `apt-get dist-upgrade` on installation, it would just force this removal.

On Debian I didn't face such problems so far, it seems to be the transfer from Debian repo to Raspbian repo that is not done too carefully by times.
€: Jep, it's just a missing version bump of the underlying library...:
```
The following packages have unmet dependencies:
 perl : Depends: libperl5.26 (= 5.26.1-4+b1) but 5.26.1-4 is to be installed
```
This is at right version on Debian repo, but yeah just forgotten to be pulled as well into Raspbian... A bid unprofessional from my point of view... I mean should be not to hard to by default pull all dependencies upgrades together with the main package you want. Or just test the installation of any package, pulled into the repo once...
Never faced such issues on Debian Buster. Everything moved into "testing" branch there is already very well tested before, so testing is basically as stable as stable, just without freezed package major versions. But this simple forgotten dependency pulls into Raspbian Buster, makes using quite dangerous (never automate apt upgrade)...
Ah enough negative words, we do dump mistakes as well 😄. Just good to know, that at least for Raspbian, "testing" means indeed "testing" and should be not used on production system, if you don't know exactly what you do.
Debian Buster on the other Hand works like a charm on my VM since months 👍.

**Comment by MichaIng** [comment]
Further Buster integration of PHP: https://github.com/Fourdee/DietPi/pull/1562

**Comment by MichaIng** [comment]
🈯️ perl upgrade issue resolved, finally the necessary library is there:
```
13 packages can be upgraded. Run 'apt list --upgradable' to see them.
2018-02-28 23:37:38 root@micha:/var/log# apt list --upgradable
Listing... Done
...
libperl5.26/testing 5.26.1-4+b1 armhf [upgradable from: 5.26.1-4]
...
```

**Comment by MichaIng** [comment]
🈯️ <s>Deluge is not available in Debian Buster repo, thus fails to install:</s> Now availabe: https://packages.debian.org/buster/deluge
🈯️ RPi: But it is available on Raspbian Buster repo 🤔.

Possible would be installation from source, but I will not touch this until Buster is moved into stable branch.
__________________
🈯️  timedatectl output changed, thus run_ntpd status fails: **=> fixed with: https://github.com/Fourdee/DietPi/pull/1661**
```
root@VM-Buster:~# timedatectl
                      Local time: Thu 2018-03-29 21:37:32 CEST
                  Universal time: Thu 2018-03-29 19:37:32 UTC
                        RTC time: Thu 2018-03-29 19:37:33
                       Time zone: Europe/Berlin (CEST, +0200)
       System clock synchronized: yes
systemd-timesyncd.service active: yes
                 RTC in local TZ: no
```

**Comment by MichaIng** [comment]
🈯️ **€: Resolved!** Apache and PHP dependency issue on Raspbian repo (again). All PHP modules have new version, just `php7.2-curl` was forgotten and would be removed. Similar for Apache2: `apache2` and `apache2-utils` are on new version, but `apache2-bin` and `apache2-data` are forgotten, which blocks upgrade or breaks Apache... I will open an issue/bugreport about that. It's just unnecessary, even that it's "testing".

For my impression, it would be totally possible to publish Buster images for x86 devices. My VM is just running totally without any issue, rock solid repo, even that you will have often many APT package updates. Not sure about other SBCs, how well their special repos work with Debian Buster, but RPi is still very unstable, actually just because incomplete APT repo upgrades.

**Comment by MichaIng** [comment]
🈴 **€: Raspbian `php-redis`, `php-igbinary` and `php-apcu` PHP7.2 compatibility lost again... WTF???**

**Comment by MichaIng** [comment]
Okay guys, for those who are interested, we uploaded some Buster **testing** images for VMware and VirtualBox: https://dietpi.com/downloads/testing/
- Jep, no official support, testing from Debian side and explicitly testing with no guarantees from our side 😉!

Would be great if some could join testing by using it, so we can fix issues now and provide official Buster images as fast as Buster is moved to stable branch (approx summer 2019) 😃.

Debian Buster generally works quite stable on one of my testing VMs, since it became new testing branch ~1 year ago. You will face a high frequency of APT and kernel updates, that'd be said!
Nearly all development I do on DietPi, I also test against Buster. But for sure I did not yet fully tried every software title in any setup combination and every feature of DietPi. So please note that there can always occur errors and if you don't mind, please report them here, so we can fix them fast.

Btw: My little production home server is Raspbian Buster, which is from repo side quite unstable in comparison to Debian (see above). Thus I cannot recommend to use it and will not provide Raspbian Buster images for now 😉.
For now we focus on the Buster Debian packages and changes and exclude custom firmware repos and hardware related incompatibilities though real SBCs.

**Comment by Fourdee** [comment]
@MichaIng 

👍 Site was updated few days ago to show optional Buster experimental images. 

Very stable. Iam yet to find an issue. 💯

**Comment by MichaIng** [comment]
Next dangerous Raspbian repo bug:
<details>

```
2018-10-13 00:14:03 root@micha:/var/log# APT
Hit:1 https://archive.raspberrypi.org/debian stretch InRelease
Hit:2 https://ftp.gwdg.de/pub/linux/debian/raspbian/raspbian buster InRelease
Reading package lists... Done
Building dependency tree
Reading state information... Done
4 packages can be upgraded. Run 'apt list --upgradable' to see them.
Listing... Done
libaprutil1-dbd-sqlite3/testing 1.6.1-3+b1 armhf [upgradable from: 1.6.1-3]
libaprutil1-ldap/testing 1.6.1-3+b1 armhf [upgradable from: 1.6.1-3]
perl-base/testing 5.26.2-7+b1 armhf [upgradable from: 5.26.2-7]
perl/testing 5.26.2-7+b1 armhf [upgradable from: 5.26.2-7]
Reading package lists... Done
Building dependency tree
Reading state information... Done
Calculating upgrade... Done
The following packages were automatically installed and are no longer required:
  apache2-data apache2-utils augeas-lenses certbot galera-3 libaio1 libapr1 libaprutil1 libaprutil1-dbd-sqlite3 libaprutil1-ldap libaugeas0 libbrotli1 libgdbm-compat4 libgdbm5 libjansson4 libjemalloc1 liblua5.2-0 libperl5.26
  libpython3-stdlib libreadline5 libwrap0 lsof mariadb-client-core-10.1 mariadb-common mariadb-server-core-10.1 perl-modules-5.26 python3 python3-acme python3-asn1crypto python3-augeas python3-certbot python3-certifi
  python3-cffi-backend python3-chardet python3-configargparse python3-configobj python3-cryptography python3-distutils python3-future python3-idna python3-josepy python3-lib2to3 python3-minimal python3-mock python3-openssl
  python3-parsedatetime python3-pbr python3-pkg-resources python3-requests python3-requests-toolbelt python3-rfc3339 python3-setuptools python3-six python3-tz python3-urllib3 python3-zope.component python3-zope.event
  python3-zope.hookable python3-zope.interface python3.6 python3.6-minimal socat
Use 'apt autoremove' to remove them.
The following packages will be REMOVED:
  apache2 apache2-bin libapache2-mod-php libapache2-mod-php7.2 libconfig-inifiles-perl libdbi-perl mariadb-client-10.1 mariadb-server mariadb-server-10.1 perl python-certbot-apache python3-certbot-apache
The following packages will be upgraded:
  perl-base
1 upgraded, 0 newly installed, 12 to remove and 0 not upgraded.
Need to get 1,251 kB of archives.
After this operation, 105 MB disk space will be freed.
Do you want to continue? [Y/n] n
Abort.
2018-10-13 00:14:25 root@micha:/var/log# apt install perl
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 perl : Depends: libperl5.26 (= 5.26.2-7+b1) but 5.26.2-7 is to be installed
E: Unable to correct problems, you have held broken packages.
2018-10-13 00:14:38 root@micha:/var/log# apt install perl-base
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following packages were automatically installed and are no longer required:
  apache2-data apache2-utils augeas-lenses certbot galera-3 libaio1 libapr1 libaprutil1 libaprutil1-dbd-sqlite3 libaprutil1-ldap libaugeas0 libbrotli1 libgdbm-compat4 libgdbm5 libjansson4 libjemalloc1 liblua5.2-0 libperl5.26
  libpython3-stdlib libreadline5 libwrap0 lsof mariadb-client-core-10.1 mariadb-common mariadb-server-core-10.1 perl-modules-5.26 python3 python3-acme python3-asn1crypto python3-augeas python3-certbot python3-certifi
  python3-cffi-backend python3-chardet python3-configargparse python3-configobj python3-cryptography python3-distutils python3-future python3-idna python3-josepy python3-lib2to3 python3-minimal python3-mock python3-openssl
  python3-parsedatetime python3-pbr python3-pkg-resources python3-requests python3-requests-toolbelt python3-rfc3339 python3-setuptools python3-six python3-tz python3-urllib3 python3-zope.component python3-zope.event
  python3-zope.hookable python3-zope.interface python3.6 python3.6-minimal socat
Use 'apt autoremove' to remove them.
Suggested packages:
  perl
The following packages will be REMOVED:
  apache2 apache2-bin libapache2-mod-php libapache2-mod-php7.2 libconfig-inifiles-perl libdbi-perl mariadb-client-10.1 mariadb-server mariadb-server-10.1 perl python-certbot-apache python3-certbot-apache
The following packages will be upgraded:
  perl-base
1 upgraded, 0 newly installed, 12 to remove and 2 not upgraded.
Need to get 1,251 kB of archives.
After this operation, 105 MB disk space will be freed.
Do you want to continue? [Y/n] n
Abort.
2018-10-13 00:14:53 root@micha:/var/log# apt install libaprutil1-ldap
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 libaprutil1-ldap : Depends: libaprutil1 (= 1.6.1-3+b1) but 1.6.1-3 is to be installed
E: Unable to correct problems, you have held broken packages.
2018-10-13 00:15:05 root@micha:/var/log# apt install libaprutil1-dbd-sqlite3
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 libaprutil1-dbd-sqlite3 : Depends: libaprutil1 (= 1.6.1-3+b1) but 1.6.1-3 is to be installed
E: Unable to correct problems, you have held broken packages.
```
</details>

- `perl` update, but required `libperl` update is missing
- `perl-base` update, but dependency update of it's dependants missing => all `perl-base` dependants will be removed on unattended apt-get dist-upgrade
- `libaprutil1` child package updates, but `libaprutil1` update itself forgotten...

Again, never faced something similar on Debian, this is a Raspbian-only issue that regularly occurs with different packages by times. Quite dangerous, if one accidentally hits yes or, as said, runs the update unattended with `-y` option.
🈯️ **€: Dependencies have been added/fixed now.**
____________
The php-apcu/redis/igbinary issue (php7.0/7.1 version, even that repo default already on php7.2) seems to at least resolved again. Hope it stays now like this.

**Comment by Fourdee** [comment]
@MichaIng 

From past experience, with RPi, wait for their official Buster image before we support it. They have their own repo, which they must iron out.

**Comment by MichaIng** [comment]
@Fourdee 
> wait for their official Buster image before we support it.

Yes in every case. I just like being on the edge (should go with Arch Linux, lol) and would like to have Buster support fully integrated into DietPi, before it's officially released to stable. So we can do concurrent release at best 😃.

For RPi official Buster images we anyway need to wait longer, until also https://archive.raspberrypi.org has Buster branch. If I remember right, this took again some time after Debian Buster release.
____________
So basically those ongoing Raspbian repo dependency issues are of course non-critical, since it's officially testing, but it makes life for testers unnecessary (?) hard for my impression. But I am complaining here all the time, now it's time to address this to the repo maintainers, being constructive 😄.

**€:** https://bugs.launchpad.net/raspbian
What the hell... Ubuntu One account needed to create a Raspbian bug report??? 😠 I am on Debian/Raspbian and stay away from Ubuntu for reason. Yeah launchpad.net is from Canonical. No idea why one would host a bug tracking system there, if not related to Ubuntu, even being kind of a rival. Not too big issue if it was a dedicated account, but I really don't like those _"create one account to connect to everything and allow everything to store to and access your merged account information"_.
Now I am going on with unproductive trolling here. Sorry for my childishness 😅.
**NB:** As fast as Microsoft connects GitHub accounts with Microsoft accounts, I am out! At least on Skype it was still possible to connect via dedicated Skype account, the last time I needed to use it, so we can hope 😄.
_____________
Bug reports that pretty much cover the main issues with Raspbian Buster:
- Due to `raspi-copies-and-fills` as mentioned in OT: https://bugs.launchpad.net/raspbian/+bug/1792818 + https://bugs.launchpad.net/raspbian/+bug/1779563
- Similar to all the other temporary repo dependency problems: https://bugs.launchpad.net/raspbian/+bug/1708779

**Comment by MichaIng** [comment]
Needed to switch to PHP7.3 to have all modules available: https://github.com/Fourdee/DietPi/pull/2259
- Possibly incompatibilities of PHP using software possible. Jep Buster is still "testing" 😉.

<s>On RPi this currently fails, since php-apcu/redis/igbinary are not yet on PHP7.3 version in Raspbian repo 🙄.</s> 🈯️ **Raspbian APCu+Redis modules on PHP7.3 now as well**

**Comment by MichaIng** [comment]
VirtualBox: TTY1 terminal resolution always resets to ~800x600 or something like this, after TTY is taken over from boot/kernel.
When commenting the `GRUB_GFXPAYLOAD_LINUX=keep` setting, then the resolution goes down to VBox default 640xsomething in between, so this is still correct/works. But when TTY1 service takes over the terminal, there seem to be an additional setting somewhere that applies 800x600. This is done regardless of grub settings, so has nothing to do with grub at all.. So far couldn't find something about it.

It happens the same time, when the console font changes. I tried `dpkg-reconfigure console-setup` with different settings, but this only affects the font, not the console itself.
- https://manpages.debian.org/jessie/console-setup/console-setup.5.en.html#OPTIONS `VIDEOMODE="1280x1024"` has no effect as well...

**Comment by MichaIng** [comment]
Last update wave on Debian+Raspbian Buster, including systemd + many libraries, breaks Dropbear. Once logged out, connecting via SSH fails. Restart of Dropbear service resolves it.

Damn, I locked out myself from my RPi with this 🤣.

**Comment by MichaIng** [comment]
🈴 Current ownCloud 10.0.10 does not support PHP7.3 => Disabled on Buster for now: https://github.com/Fourdee/DietPi/pull/2361/commits/a454034c0060f815e88deee8048bb89404b43ac2
_________
~🈴 New dependency issue on Raspbian Buster: Python3 upgraded to v3.7, but python-certbot(-apache|-nginx)? not yet updated accordingly and would be removed for this reason, if python uprgade was forced.~ **€: Resolved**

**Comment by MichaIng** [comment]
Debian Buster release soon. Feature/Package freeze schedule has started: https://release.debian.org/testing/freeze_policy.html
> 2019-01-12 - Transition freeze
> Starting 2019-01-12, new transitions and large/disruptive changes are no longer acceptable for buster.
> 2019-02-12 - Soft-freeze
> 2019-03-12 - Full-freeze

With v6.21 I will put some effort in testing remaining software titles on Buster. Especially some 3rd party repos don't have a Buster branch yet, so we need to force Stretch there.

Raspbian Buster release will be later and according to the many above critical repo issues, this is required.

**Comment by MichaIng** [comment]
Damn, the soft freeze did not reduce the amount of APT package updates, but increased it, in some cases with critical major version updates, e.g. mariadb v10.1 => v10.3.

One single day and 43 updated packages (and I have ONLY installed Nextcloud), including systemd, udev, APT itself, pathon, PHP...
```
43 packages can be upgraded. Run 'apt list --upgradable' to see them.
Listing... Done
apt-transport-https/testing 1.8.0~rc2 all [upgradable from: 1.8.0~beta1]
apt-utils/testing 1.8.0~rc2 armhf [upgradable from: 1.8.0~beta1]
apt/testing 1.8.0~rc2 armhf [upgradable from: 1.8.0~beta1]
ca-certificates/testing 20190110 all [upgradable from: 20170717]
libapache2-mod-php7.3/testing 7.3.1-3 armhf [upgradable from: 7.3.1-1]
libapt-inst2.0/testing 1.8.0~rc2 armhf [upgradable from: 1.8.0~beta1]
libapt-pkg5.0/testing 1.8.0~rc2 armhf [upgradable from: 1.8.0~beta1]
libargon2-1/testing 0~20171227-0.2 armhf [upgradable from: 0~20171227-0.1]
libdb5.3/testing 5.3.28+dfsg1-0.3 armhf [upgradable from: 5.3.28+dfsg1-0.2]
libgcrypt20/testing 1.8.4-5 armhf [upgradable from: 1.8.4-4]
libgnutls30/testing 3.6.6-2 armhf [upgradable from: 3.6.5-2+rpi1]
libhogweed4/testing 3.4.1-1 armhf [upgradable from: 3.4.1~rc1-1]
libjson-c3/testing 0.12.1+ds-2 armhf [upgradable from: 0.12.1-1.3]
liblzma5/testing 5.2.4-1 armhf [upgradable from: 5.2.2-1.3]
libmpfr6/testing 4.0.2-1 armhf [upgradable from: 4.0.2~rc1-1]
libnettle6/testing 3.4.1-1 armhf [upgradable from: 3.4.1~rc1-1]
libnghttp2-14/testing 1.36.0-2 armhf [upgradable from: 1.36.0-1]
libparted2/testing 3.2-24 armhf [upgradable from: 3.2-23]
libpcre2-8-0/testing 10.32-4 armhf [upgradable from: 10.32-3]
libpng16-16/testing 1.6.36-5 armhf [upgradable from: 1.6.36-2]
libpopt0/testing 1.16-12 armhf [upgradable from: 1.16-11]
libslang2/testing 2.3.2-2 armhf [upgradable from: 2.3.2-1+b1]
libsystemd0/testing 240-5+rpi1 armhf [upgradable from: 240-4+rpi1]
libudev1/testing 240-5+rpi1 armhf [upgradable from: 240-4+rpi1]
parted/testing 3.2-24 armhf [upgradable from: 3.2-23]
php7.3-cli/testing 7.3.1-3 armhf [upgradable from: 7.3.1-1]
php7.3-common/testing 7.3.1-3 armhf [upgradable from: 7.3.1-1]
php7.3-curl/testing 7.3.1-3 armhf [upgradable from: 7.3.1-1]
php7.3-gd/testing 7.3.1-3 armhf [upgradable from: 7.3.1-1]
php7.3-intl/testing 7.3.1-3 armhf [upgradable from: 7.3.1-1]
php7.3-json/testing 7.3.1-3 armhf [upgradable from: 7.3.1-1]
php7.3-mbstring/testing 7.3.1-3 armhf [upgradable from: 7.3.1-1]
php7.3-mysql/testing 7.3.1-3 armhf [upgradable from: 7.3.1-1]
php7.3-opcache/testing 7.3.1-3 armhf [upgradable from: 7.3.1-1]
php7.3-readline/testing 7.3.1-3 armhf [upgradable from: 7.3.1-1]
php7.3-xml/testing 7.3.1-3 armhf [upgradable from: 7.3.1-1]
php7.3-zip/testing 7.3.1-3 armhf [upgradable from: 7.3.1-1]
python3-pkg-resources/testing 40.7.1-1 all [upgradable from: 40.6.3-1]
python3-setuptools/testing 40.7.1-1 all [upgradable from: 40.6.3-1]
systemd-sysv/testing 240-5+rpi1 armhf [upgradable from: 240-4+rpi1]
systemd/testing 240-5+rpi1 armhf [upgradable from: 240-4+rpi1]
udev/testing 240-5+rpi1 armhf [upgradable from: 240-4+rpi1]
xz-utils/testing 5.2.4-1 armhf [upgradable from: 5.2.2-1.3]
```

To all who run DietPi on Buster:
- Carefully watch the update process, better run `apt dist-upgrade` more often then less, to not have hundreds of packages upgraded in a single run.
- Report any issues you face, e.g.: https://github.com/Fourdee/DietPi/pull/2490
- We will assist to get those fixed quickly.

**Comment by MichaIng** [comment]
Yey, archive.raspberrypi.org repo already has a Buster branch: https://archive.raspberrypi.org/debian/dists/
It simply does not provide `raspi-copies-and-fills` anymore, which was incompatible with Buster udev version (see OT).

Besides that: No kernel or other package update when switching from the Stretch repo currently.
Will remove the s/buster/stretch/g sed when applying the repo and patch on v6.22 für RPi Buster systems (although I doubt there are much or any, besides mine 😄.

DietPi-Set_Software: https://github.com/Fourdee/DietPi/commit/1f98515dc8443c5a0a08bd1cb56e575b04960fea
DietPi-Patch: https://github.com/Fourdee/DietPi/commit/e59cc70df2682cb013a693e0d126aeb3b7e7d762

**Comment by MichaIng** [comment]
~🈴 RPi `i-sable-k2m` module build might fail due to `gcc` version mismatch: https://github.com/Fourdee/DietPi/issues/2547#issuecomment-463831296~
#### EDIT: Driver is now kernel built-in since v4.19 as dtoverlay `i-sable-q2m`.

**Comment by MichaIng** [comment]
Okay I mark this issue as closed now. We will gradually roll out Buster images for additional devices, at first as experimental/beta images, after official Debian (respectively Raspbian) Buster release and some user feedback, as new official image.

There are still some issues opened but those can be handled individually: https://github.com/MichaIng/DietPi/issues?q=is%3Aopen+is%3Aissue+label%3A"Debian+Buster"
